test/python/providers/test_options.py

```diff
@@ -105,3 +108,87 @@ def test_hasattr(self):
         options = Options(shots=1024)
         self.assertTrue(hasattr(options, "shots"))
         self.assertFalse(hasattr(options, "method"))
+
+
+class TestOptionsSimpleNamespaceBackwardCompatibility(QiskitTestCase):
+    """Tests that SimpleNamespace-like functionality that qiskit-experiments relies on for Options
+    still works."""
+
+    def test_unpacking_dict(self):
+        kwargs = {"hello": "world", "a": "b"}
+        options = Options(**kwargs)
+        self.assertEqual(options.__dict__, kwargs)
+        self.assertEqual({**options.__dict__}, kwargs)
+
+    def test_setting_attributes(self):
+        options = Options()
+        options.hello = "world"
+        options.a = "b"
```

Comment:
`test/python/providers/test_options.py:125:8: E0237: Assigning to attribute 'hello' not defined in class slots (assigning-non-slot)` from https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=33135&view=logs&j=f0ab2c38-58fa-55ad-8e8b-d3984171f737&t=980625fc-4a64-50e4-7ca0-679c2a4bbda8&l=15

Reply:
Fixed by suppressing it in bdcdf64.  The test succeeds, and pylint could realise that this is possible!